api/grpcserver/smesher_service.go

```diff
@@ -25,40 +27,34 @@ func (s SmesherService) RegisterService(server *Server) {
 }
 
 // NewSmesherService creates a new grpc service using config data.
-func NewSmesherService(miner api.MiningAPI) *SmesherService {
-	return &SmesherService{
-		Mining: miner,
-	}
+func NewSmesherService(post api.PostAPI, smeshing api.SmeshingAPI) *SmesherService {
+	return &SmesherService{post, smeshing}
 }
 
 // IsSmeshing reports whether the node is smeshing
 func (s SmesherService) IsSmeshing(ctx context.Context, in *empty.Empty) (*pb.IsSmeshingResponse, error) {
 	log.Info("GRPC SmesherService.IsSmeshing")
 
-	stat, _, _, _ := s.Mining.MiningStats()
-	isSmeshing := stat == activation.InitDone
-	return &pb.IsSmeshingResponse{IsSmeshing: isSmeshing}, nil
+	return &pb.IsSmeshingResponse{IsSmeshing: s.smeshing.Smeshing()}, nil
 }
 
 // StartSmeshing requests that the node begin smeshing
 func (s SmesherService) StartSmeshing(ctx context.Context, in *pb.StartSmeshingRequest) (*pb.StartSmeshingResponse, error) {
 	log.Info("GRPC SmesherService.StartSmeshing")
 
+	// TODO(moshababo): remove DataDir and CommitmentSize from the request proto definition.
```

Comment:
Please open an issue for this here: https://github.com/spacemeshos/api/issues

Reply:
https://github.com/spacemeshos/api/issues/114